bluebottle/bb_tasks/views.py

```diff
@@ -79,6 +83,21 @@ def get_queryset(self):
             qs = qs.filter(Q(title__icontains=text) |
                            Q(description__icontains=text))
 
+        # Searching for tasks can mean 2 things:
+        # 1) Search for tasks a specifc day
+        # 2) Search for tasks that take place over a period of time
+        start_date = self.request.query_params.get('start_date', None)
+        end_date = self.request.query_params.get('end_date', None)
+
+        # User searches for tasks on a specific day.
+        if start_date and not end_date:
+            qs = qs.filter(Q(type='event', deadline=start_date) |
+                           Q(type='ongoing', deadline__gte=start_date))
+
+        # User searches for tasks in a specific range
+        if start_date and end_date:
+            qs = qs.filter(deadline__range=[start_date, end_date])
```

Comment:
Which line are you referring to? Because line 99 doesn't filter on type and if there is an explicit end_date it shouldn't return anything outside of that range.